Add new entries to clarify multicluster terminology. (#7591)
istio/istio.io: content/en/docs/reference/glossary/primary-cluster.md

---
title: Primary Cluster
test: n/a
---

> A primary cluster is a cluster with a control plane, which observes all

Reviewer: Can/should we link to the glossary entry for control plane here?

Reviewer 2: I personally think the term “primary” is too vague. I prefer a name which is a bit more intention revealing such as “control cluster”.

Reviewer 3: I tend to agree "primary" is too vague, and so is "remote cluster" for that matter. Neither term really aligns very well with what they are. A primary cluster is a "self-controlled cluster" and a remote cluster is an "externally-controlled cluster" (the external control can come from the control plane in a self-controlled cluster or a control plane running outside of any cluster). I'm not sure that "self-controlled" and "externally-controlled" are very good names to use, but the point is that "primary" and "remote" don't really seem to be indicative of what they are.

> A primary cluster is a cluster with a control plane, which observes all
> clusters in the mesh for endpoints and facilitates cross-cluster load balancing

Reviewer 3: Is this correct?
```suggestion
A primary cluster is a cluster with a control plane, which observes more than one
cluster in a multicluter mesh for endpoints and facilitates cross-cluster load balancing
```

Author: @nmittler @smawson WDYT?

Reviewer 4: I think the original text was correct. A single-cluster mesh consists of a single primary cluster.

Reviewer 3: I was worried about the other end, "observes all clusters". Is that always true? Isn't it possible to have a primary cluster that only observes a subset of the clusters in the mesh, while another primary cluster handles others?

Reviewer 4: Not in any documented configuration that I'm aware of. Unless you're talking about versioned control planes, which is slightly orthogonal.

Reviewer 2: @frankbu is correct. A single istiod control plane can be configured to observe only a subset of a cluster.

Also the statement “which observes more than one cluster” is too strict. Is should be “which observes resources from one or more clusters...”.

Reviewer 4: @dcberg 

> @frankbu is correct. A single istiod control plane can be configured to observe only a subset of a cluster.

Just to be clear, are you referring to control plane revisions?

If we want to be more precise, maybe we can rephrase to something like:

```
A primary cluster is a cluster within a mesh that contains a control plane. The control plane observes configuration resources from a single source (e.g. Kubernetes API Server). It also observes service endpoints from one or more clusters in the mesh in order to facilitate cross-cluster load balancing between workloads.
```

Reviewer 3: > Multiple primary clusters can be connected together

I don't think what "connected together" means is clear. What's it trying to say? You can have more than one (synchronized) primary cluster in the same mesh for HA? I think it needs to say something more specific.

> between clusters.

Reviewer: Things we should mention:
- Multiple primary clusters can be connected together in a single mesh (point to glossary entry for mesh)
- Primary clusters can act as the control plane for remote clusters (point to glossary entry for remote cluster).

I think the first one can replace the note about observing endpoints and facilitating cross-cluster LB, which seems a bit too much implementation detail of how they are connected, and not really part of the definition of a primary cluster.
